source/common/http/http2/codec_impl.cc

```diff
@@ -1120,7 +1090,23 @@ Status ConnectionImpl::sendPendingFrames() {
     }
     RETURN_IF_ERROR(sendPendingFrames());
   }
-  return okStatus();
+
+  // After all pending frames have been written into the outbound buffer check if any of
+  // protocol constraints had been violated.
+  Status status = protocol_constraints_.checkOutboundFrameLimits();
+  if (!status.ok()) {
+    ENVOY_CONN_LOG(debug, "error sending frames: Too many frames in the outbound queue.",
+                   connection_);
+  }
+  return status;
+}
+
+bool ConnectionImpl::sendPendingFramesAndHandleError() {
+  if (!sendPendingFrames().ok()) {
+    scheduleProtocolConstraintViolationCallback();
+    return true;
```

Comment:
returning true on error is a bit unusual, but you are documenting the return value so it seems fine.